src/main/java/htsjdk/variant/vcf/VCFCompoundHeaderLine.java

```diff
@@ -240,7 +240,7 @@ protected VCFCompoundHeaderLine(String line, VCFHeaderVersion version, Supported
             throw new TribbleException(mapping.get("Type") + " is not a valid type in the VCF specification (note that types are case-sensitive)");
         }
         if (type == VCFHeaderLineType.Flag && !allowFlagValues())
-            throw new IllegalArgumentException("Flag is an unsupported type for this kind of field");
+            throw new IllegalArgumentException("Flag is an unsupported type for this kind of field at line number "+ lineNumber);
```

Comment:
I like the idea of reporting line numbers, but ideally we wouldn't have to propagate them to the line constructors like this. For this issue it's probably sufficient to just include the text of the actual line (which is available in `line`) in the error message.

Reply:
Thank you for the feedback. I made the suggested changes. Please let me know if you find anything else.